### DISCUSSION TITLE
where is the settings for adding local instance url

### DISCUSSION THREAD

**Discussion by charles-prof** [author-more/penpot-desktop#46] [Q&A] [answered]
I downloaded penpot desktop on windows

used the docker compose to pull and run penpot locally ( self-host )

docker successfully running and penpot frontend is running on 9001

![image](https://github.com/user-attachments/assets/67cad906-80b7-4b84-a353-06aac80bcf30)

[where is the gear icon mentioned in the wiki](https://github.com/author-more/penpot-desktop/wiki/Self%E2%80%90hosting)

**Comment by Belar** [comment]
Hi @charles-prof,
it all sounds good. The gear icon button should be visible in the top right corner, next to the window controls (minimise, close etc.) - like on the screenshot in the wiki.
However, if it’s the UI acting up (there was a similar issue in the past, patched since then), please try a restart of the app.

**Reply by charles-prof** [reply]
Thank you for your quick reply.

After restart. now i see the icon but i'm stuck with 'no response' from server

![image](https://github.com/user-attachments/assets/722c7507-0ef5-4b3a-ba58-ba727aa2fb4f)

docker compose running successfully but no response from the endpoint. tried in app, browser and curl

![image](https://github.com/user-attachments/assets/a1edfe05-33b9-432a-a87d-3d3932377f37)

**Reply by Belar** [reply]
It looks like an issue outside the Penpot Desktop.

I don’t use Windows, but recall Docker behaved differently there when it came to port mapping and localhost, because of the virtual machine. I don’t know if it’s still the case. If you would like to do some troubleshooting, I would search around “windows docker desktop” and “localhost no connection”. You could also ask the Penpot community, someone is probably self-hosting Penpot on Windows.

**Reply by charles-prof** [reply]
I'll try and pingback

**Reply by charles-prof** [reply]
ok. the problem could be an issue in port mapping.

what port does the frontend run on ? ( it is mapped from 8080 to 9001 in docker-compose.yml )

I tried something with AI help to inspect exposed port ( which could be wrong )

```
C:\Users\Games\penpot>docker inspect -f '{{.Config.ExposedPorts}}' penpotapp/frontend:latest
'map[80/tcp:{}]'
```

when i changed the port mapping to 80 instead of 8080, the response was received and the ui said retry - some selection is null errors in console

after few gimmicks, it works. 

The answer is port 80

**Reply by Belar** [reply, marked as answer]
Thank you for the update.

I have an old `docker-compose.yml`, with the `9001:80` mapping, and thought the `8080` is a Docker Desktop for Windows thing (it has an extra layer of a Linux VM). I took a look at the Penpot repo now and the setup has changed to `8080` recently, but there is [a problem with images](https://github.com/penpot/penpot/issues/5430). Hence the issues you run into.

**Reply by charles-prof** [reply]
That's ok. i'm glad that i solved this with help of an AI in the meantime. Your support and penpot project is really helpful. keep up
